File: 2022/0627/PS_구명보트_김주은.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        Arrays.sort(people);
        
        int left = 0;
        int right = people.length-1;
        
        while(left<right){
            if(people[right]+people[left]<=limit){
                ++answer;
                ++left;
                --right;
            } else{
                ++answer;
                --right;
            }
        }
        
        if(left==right){
            ++answer;
        }
        
        return answer;
    }
}

File: 2023/0214구명보트/PS_구명보트_김주은.py
def solution(people, limit):
    people.sort()
    
    start=0
    end=len(people)-1
    
    answer=0
    
    while start<=end:
        if people[start]+people[end]<=limit:
            start+=1
        answer+=1
        end-=1
    
    return answer
